calendar_resource/models/calendar_event.py

```diff
@@ -34,12 +114,141 @@ def _check_resource_ids_dbl_book(self):
+                        )
+                        % (
+                            resource.name,
+                            ', '.join([
+                                categ.name for categ in categs
+                            ]),
+                        )
+                    )
+
+    @api.multi
+    @api.constrains('resource_ids', 'start', 'stop')
+    def _check_resource_ids_leaves(self):
+
+        for record in self:
+
+            if record._event_in_past():
+                continue
+
+            for resource in record.resource_ids.filtered(
+                    lambda s: s.calendar_id and s.calendar_id.leave_ids):
+
+                conflict_leaves = resource.calendar_id.leave_ids.filtered(
+                    lambda s: s.date_from < record.stop and
+                    s.date_to > record.start
+                )
+
+                if not conflict_leaves:
+                    continue
+
+                datetimes = [(c.date_from, c.date_to) for c in conflict_leaves]
+                raise ValidationError(
+                    _(
+                        "The resource, '%s', is on leave during "
+                        "the following times which are conflicting with "
+                        "this event.\n%s"
+                    )
+                    % (
+                        resource.name,
+                        self._format_datetime_interval_list(datetimes),
+                    )
+                )
+
+    @api.multi
+    @api.constrains('resource_ids', 'start', 'stop')
+    def _check_resource_ids_working_times(self):
+
+        for record in self:
+
+            if record._event_in_past():
+                continue
+
+            event_start = fields.Datetime.from_string(record.start)
+            event_stop = fields.Datetime.from_string(record.stop)
+            event_days = record._get_event_date_list()
+
+            for resource in record.resource_ids.filtered(
+                    lambda s: s.calendar_id):
+
+                available_intervals = []
+                conflict_intervals = []
+
+                for day in event_days:
+
+                    datetime_start = datetime.combine(day, time(00, 00, 00))
+                    datetime_end = datetime.combine(day, time(23, 59, 59))
+
+                    intervals = \
+                        resource.calendar_id.get_working_intervals_of_day(
+                            start_dt=datetime_start,
+                            end_dt=datetime_end,
+                            resource_id=resource.id,
+                        )
+
+                    if not intervals and record.allday:
+                        conflict_intervals.append(
+                            (datetime_start, datetime_end)
+                        )
+                    else:
+                        available_intervals += intervals
+
+                if not record.allday:
+                    conflict_intervals = self.env['resource.calendar'].\
```

Comment:
Thanks for mentioning. Not being tested pointed to a detail I missed in tests which needs to be fixed.